nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardWebClientService.java

```diff
@@ -87,17 +98,16 @@ public void setConnectTimeout(final Duration connectTimeout) {
      */
     public void setReadTimeout(final Duration readTimeout) {
         Objects.requireNonNull(readTimeout, "Read Timeout required");
-        okHttpClient = okHttpClient.newBuilder().readTimeout(readTimeout).build();
+        this.readTimeout = readTimeout;
     }
 
     /**
-     * Set timeout for writing requests to socket connection
+     * Set timeout for writing requests to socket connection is not supported with current HTTP Client implementation
```

Comment:
It does seem possible to set a timeout on requests

https://www.baeldung.com/java-httpclient-timeout

I dont quite follow in the new api read/write timeout but it does seem like you can set a timeout on the requests.

Reply:
Thanks @joewitt. The changes already included handling of the `readTimeout` property, passing it to the `HttpRequest.Builder`. However, for the sake of completeness, I added handling for `writeTimeout`, but noted that `readTimeout` is preferred. OkHttp had separate methods with different timeout behavior, but the latest commit optionally uses `writeTimeout` for the request.